frontends/p4/sideEffects.cpp

```diff
@@ -22,14 +22,19 @@ namespace P4 {
 
 namespace {
 
-// Data structure used for making explicit the order of evaluation of
-// sub-expressions in an expression.
-// An expression e will be represented as a sequence of temporary declarations,
-// followed by a sequence of statements 'statements' (mostly assignments to the temporaries,
-// but which could also include conditionals for short-circuit evaluation).
+/** Data structure used for making the order of evaluation explicit for
+ * sub-expressions in an expression.  An expression ```e``` will be represented
+ * as a sequence of temporary declarations, followed by a sequence of
+ * statements (mostly assignments to the temporaries, but which could also
+ * include conditionals for short-circuit evaluation).
+ */
 struct EvaluationOrder {
     ReferenceMap* refMap;
+
+    /// Output: The original expression with side-effecting sub-expressions
```

Comment:
An expression whose evaluation will produce the same result as the original one, but where side-effects have been factored out such that each statement produces at most one side-effect.